htmresearch/encoders/language_encoder.py

```diff
@@ -183,11 +183,14 @@ def sparseUnion(self, counts):
 
     @param counts     (Counter)   A count of the ON bits for the union bitmap.
 
-    @return           (list)      A sparsified union bitmap.
+    @return           (tuple)      A sparsified union bitmap.
     """
+    positionsSortedByCount = tuple(x[1] for x in counts.getNonZerosSorted())
```

Comment:
The previous implementation uses `heapq.nlargest()` (see https://github.com/python/cpython/blob/master/Lib/collections/__init__.py#L548) which, I think, is not stable.

I will try adding a small random number and see if there's not too much of a performance hit.